src/components/Menu/index.web.tsx

```diff
@@ -119,6 +119,11 @@ export function Trigger({children, label, role = 'button'}: TriggerProps) {
   } = useInteractionState()
   const {state: focused, onIn: onFocus, onOut: onBlur} = useInteractionState()
 
+  const prevControlIsOpen = React.useRef(control.isOpen)
```

Comment:
Use state for this instead, it's not legit to modify refs during render.

https://react.dev/reference/react/useState#storing-information-from-previous-renders

We have a bunch of code like this, look for `prev`